wire-ios-request-strategy/Sources/Request Strategies/Conversation/MLSConversationParticipantsService.swift

```diff
@@ -0,0 +1,141 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireDataModel
+
+// sourcery: AutoMockable
+protocol MLSConversationParticipantsServiceInterface: ConversationParticipantsServiceInterface {}
+
+class MLSConversationParticipantsService: MLSConversationParticipantsServiceInterface {
+
+    // MARK: - Properties
+
+    private let context: NSManagedObjectContext
+    private let clientIDsProvider: MLSClientIDsProviding
+
+    // MARK: - Life cycle
+
+    init(
+        context: NSManagedObjectContext,
+        clientIDsProvider: MLSClientIDsProviding? = nil
+    ) {
+        self.context = context
+        self.clientIDsProvider = clientIDsProvider ?? MLSClientIDsProvider()
+    }
+
+    // MARK: - Interface
+
+    func addParticipants(
+        _ users: [ZMUser],
+        to conversation: ZMConversation,
+        completion: @escaping AddParticipantAction.ResultHandler
+    ) {
+        Logging.mls.info("adding \(users.count) participants to conversation (\(String(describing: conversation.qualifiedID)))")
+
+        guard
+            let mlsService = getMLSService(fromSyncContext: context.zm_sync),
```

Comment:
I find it a bit weird, that contexts have weak references to each other. I wonder why this approach was chosen.
Wouldn't it be better to ensure in `init`, that the context "is a sync context" instead of getting another context through the injected context?

Reply:
I agree it would be better to be on the correct context. I've decided not to change this yet because of time constraints.